### ISSUE TITLE
LinkReferenceDefinition has null sourceSpans if title included

### ISSUE THREAD

**Issue by tjpalmer** [commonmark/commonmark-java#292]
Steps to reproduce the problem (provide example Markdown if applicable):

```
[label][1]

[1]: #not-code "Text"
```

And I've set `IncludeSourceSpans.BLOCKS_AND_INLINES` when building the parser.

Expected behavior:

The LinkReferenceDefinition has a correct sourceSpans set.

Actual behavior:

The sourceSpans is null on the LinkReferenceDefinition instance.

Note that this alternative without a title *does* have valid sourceSpans on the LinkReferenceDefinition:

```
[label][1]

[1]: #not-code
```

Using "org.commonmark:commonmark:0.21.0".

**Comment by robinst** [comment]
Thanks, fixed!

**Comment by robinst** [comment]
Released in 0.22.0 now: https://github.com/commonmark/commonmark-java/blob/commonmark-parent-0.22.0/CHANGELOG.md#0220---2024-03-15
